src/prototype_rust_sdk/workflow_context.rs

```diff
@@ -75,101 +81,97 @@ impl WfContext {
-        })
+        });
+        self.send(
+            CommandCreateRequest {
+                cmd: opts.to_command(seq).into(),
+                unblocker,
+            }
+            .into(),
+        );
+        cmd
     }
 
     /// Request to start a child workflow, Returned future resolves when the child has been started.
     pub fn start_child_workflow(
         &mut self,
         a: StartChildWorkflowExecution,
     ) -> impl Future<Output = ChildWorkflowStartStatus> {
-        let (cmd, unblocker) = WFCommandFut::new();
-        self.send(
-            CommandCreateRequest {
-                cmd: a.into(),
-                unblocker,
-            }
-            .into(),
-        );
-        cmd.map(|ue| {
+        let seq = self.next_sequence_number.fetch_add(1, Ordering::SeqCst);
+        let (cmd, unblocker) = WFCommandFut::new(Some(CommandID::ChildWorkflowStart(seq)), |ue| {
             if let UnblockEvent::WorkflowStart(_, result) = ue {
                 *result
             } else {
                 panic!("Wrong unblock event")
             }
-        })
-    }
-
-    /// Wait on completion of a child workflow execution started with `start_child_workflow`
-    pub fn child_workflow_result(
-        &mut self,
-        workflow_id: String,
-    ) -> impl Future<Output = ChildWorkflowResult> {
-        let (cmd, unblocker) = WFCommandFut::new();
+        });
         self.send(
-            CommandSubscribeChildWorkflowCompletion {
-                workflow_id,
+            CommandCreateRequest {
+                cmd: a.into(),
                 unblocker,
             }
             .into(),
         );
-        cmd.map(|ue| {
+        cmd
+    }
+
+    /// Wait on completion of a child workflow execution started with `start_child_workflow`
+    pub fn child_workflow_result(&mut self, seq: u32) -> impl Future<Output = ChildWorkflowResult> {
+        let (cmd, unblocker) = WFCommandFut::new(None, |ue| {
             if let UnblockEvent::WorkflowComplete(_, result) = ue {
                 *result
             } else {
                 panic!("Wrong unblock event")
             }
-        })
-    }
-
-    /// Cancel a timer
-    pub fn cancel_timer(&self, timer_id: &str) {
-        self.send(RustWfCmd::CancelTimer(timer_id.to_string()))
-    }
-
-    /// Cancel activity
-    pub fn cancel_activity(&self, activity_id: &str) {
-        self.send(RustWfCmd::CancelActivity(activity_id.to_string()))
+        });
+        self.send(CommandSubscribeChildWorkflowCompletion { seq, unblocker }.into());
+        cmd
     }
 
-    /// Cancel child workflow
-    pub fn cancel_child_workflow(&self, child_workflow_id: &str) {
-        self.send(RustWfCmd::CancelChildWorkflow(
-            child_workflow_id.to_string(),
-        ))
+    /// Cancel any cancellable operation
+    pub fn cancel<T: 'static>(&self, fut: &(impl Future<Output = T> + 'static)) {
+        let fut = fut as &dyn Any;
+        match fut.downcast_ref::<WFCommandFut<T>>() {
```

Comment:
This shouldn't be this way. If the `fut` argument should only ever be a `WFCommandFut` then that can be enforced in the type signature.

Reply:
I didn't want the other commands to return a `WFCommandFut`, it's not exported and I don't know if it should be.
WDYT?

Reply:
I am fine with exposing it. Then we can just make a `.cancel` on the `WFCommandFut` itself too which is possibly more desirable.

Reply:
Also desirable, combined with the generic changes to `WFCommandFut`, it would be possible to only expose `.cancel` on types that actually make sense to cancel